### PR TITLE
fix(amplify-provider-awscloudformation): Stack delete condition

### PR DESCRIPTION
**Pull request by arseneoaa**
Currently when updating/deleting the stack both authParamsJson and unauthParamsJson keep their previous condition statement.
When the new stack is created the previous values for 'Condition': {'StringEquals': {'cognito-identity.amazonaws.com:aud': idpId},' for example do not get updated.

The fix is just about deleting the condition on the object inside the Statement array instead of on the Statement array itself.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by UnleashedMind** [comment]
The fix looks good.
Could you add e2e test for the PR? Sample e2e tests can be found in the amplify-e2e-tests package. 
I'm happy offer assistance if needed. Just send an email to amplify-cli@amazon.com with subject Re: PR#4465.

**Comment by UnleashedMind** [comment]
I added the e2e tests

File: packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts [hunk 1]
```diff
@@ -27,3 +27,49 @@ export const toBeIAMRoleWithArn = async (roleName: string, arn?: string) => {
...
+    pass = false;
+    message = 'IAM GetRole threw Error: ' + e.message;
+  }
+
+  const result = {
+    message: () => message,
+    pass,
+  };
+
+  return result;
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts:74]: Small nit: I would return the object directly

```ts
  return {
    message: () => message,
    pass,
  };
```

File: packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts [hunk 2]
```diff
@@ -27,3 +27,49 @@ export const toBeIAMRoleWithArn = async (roleName: string, arn?: string) => {
   };
   return result;
 };
+
+export const toHaveAssumeRolePolicyConditionMatchingIdpId = async (roleName: string, idpId: string) => {
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts:31]: Nit: I think something like `toHaveValidAssumeRolePolicyCondition` would be easier to understand at a glance.

File: packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts [hunk 3]
```diff
@@ -27,3 +27,49 @@ export const toBeIAMRoleWithArn = async (roleName: string, arn?: string) => {
...
+      } else {
+        return false;
+      }
+    });
+
+    if (pass) {
+      message = 'Found Matching Condition';
+    } else {
+      message = 'Matching Condition does not exist';
+    }
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/aws-matchers/iamMatcher.ts:63]: Simplifies to 
```ts
message = pass ?  'Found Matching Condition' : 'Matching Condition does not exist';
```

**Comment by arseneoaa** [comment]
Thanks a lot @jhockett .
Sorry I did not have the time to look into how to do the e2e tests.

**Pull request merged**